pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -245,8 +245,7 @@ let
     (enableFeature doBenchmark "benchmarks")
     "--enable-library-vanilla"  # TODO: Should this be configurable?
     (enableFeature enableLibraryForGhci "library-for-ghci")
-  ] ++ optionals (enableDeadCodeElimination && (lib.versionOlder "8.0.1" ghc.version)) [
-     "--ghc-option=-split-sections"
+    (enableFeature enableDeadCodeElimination "split-sections")
   ] ++ optionals dontStrip [
     "--disable-library-stripping"
     "--disable-executable-stripping"
```

Comment:
Sounds good to me, maybe in a separate PR?

Reply:
Good question. We could also merge all of this into `staging` immediately and wait for it to propagate back to us via `staging-next` and `master`. Otherwise I'd say bundle everything in a single PR targeting `haskell-updates`.